### PR TITLE
redo (ticdc): set safeMode to true when redo apply

### LINKED ISSUE #7526: [CDC:ErrMySQLTxnError]MySQL txn error: Error 1062: Duplicate entry error if safe-mode is true during redo apply
### What did you do?

1. Do cdc redo log apply when safe-mode is on

### What did you expect to see?

redo log apply should succeed

### What did you see instead?

redo log failed with error: [CDC:ErrMySQLTxnError]MySQL txn error: Error 1062: Duplicate entry '23445' for key 'sbtest6.PRIMARY'

### Versions of the cluster

master, v6.4.0-alpha

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: close #7526 

#### Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
None
```

### CODE REVIEW

File: pkg/cmd/redo/apply.go [hunk 1]
```diff
@@ -38,6 +41,25 @@ func (o *applyRedoOptions) addFlags(cmd *cobra.Command) {
...
 
+func (o *applyRedoOptions) complete(cmd *cobra.Command) error {
+	// parse sinkURI as a URI
+	sinkURI, err := url.Parse(o.sinkURI)
+	if err != nil {
+		return cerror.WrapError(cerror.ErrSinkURIInvalid, err)
+	}
+	rawQuery := sinkURI.Query()
+	if rawQuery.Get("safe-mode") != "true" {
+		cmd.Println("Warning: sink-uri is not in safe-mode, " +
```
Comment (zhaoxinyu) [inline pkg/cmd/redo/apply.go:52]:
```suggestion
		cmd.Println("Warning: safe-mode is not in sink-uri, " +
```

**Review by nongfushanquan — APPROVED** [review]

File: pkg/cmd/redo/apply.go [hunk 2]
```diff
@@ -38,6 +41,25 @@ func (o *applyRedoOptions) addFlags(cmd *cobra.Command) {
...
 
+func (o *applyRedoOptions) complete(cmd *cobra.Command) error {
+	// parse sinkURI as a URI
+	sinkURI, err := url.Parse(o.sinkURI)
+	if err != nil {
+		return cerror.WrapError(cerror.ErrSinkURIInvalid, err)
+	}
+	rawQuery := sinkURI.Query()
+	if rawQuery.Get("safe-mode") != "true" {
+		cmd.Println("Warning: safe-mode is not in sink-uri, " +
```
Comment (zhaoxinyu) [inline pkg/cmd/redo/apply.go:52]: This warning is not accurate if `safe-mode=false` is in sink-uri. Should it be replaced with `Warning: safe-mode is not in sink-uri or the value is false` ?

**Review by zhaoxinyu — APPROVED** [review]

File: pkg/cmd/redo/apply.go [hunk 3]
```diff
@@ -16,7 +16,10 @@ package redo
 import (
 	"github.com/pingcap/tiflow/pkg/applier"
 	cmdcontext "github.com/pingcap/tiflow/pkg/cmd/context"
+	cerror "github.com/pingcap/tiflow/pkg/errors"
 	"github.com/spf13/cobra"
+
```
Comment (hi-rustin) [inline pkg/cmd/redo/apply.go:21]: Please refine your imports order.

**Review by hi-rustin — APPROVED** [review]

**Comment by hi-rustin** [comment]
Thanks for working on this! :shipit:

**Pull request merged**